overlord/snapstate/handlers.go

```diff
@@ -1332,137 +1343,6 @@ func triggeredMigration(oldBase, newBase string, opts *dirMigrationOptions) migr
 	return none
 }
 
-func (m *SnapManager) doHiddenMigration(t *state.Task, st *state.State, snapsup *SnapSetup, newInfo *snap.Info) (undo func(), err error) {
```

Comment:
Based on our conversation about errors types in the handlers, I think these helpers were too defensive and complicated. I removed all of them and replaced them (in doCopySnapData and doUnlinkCurrentSnap) with just calling the backend and setting the flags in SnapSetup. If anything fails, we don't try to recover since it's likely that further backend ops would also fail